packages/apputils/src/sessioncontext.tsx

```diff
@@ -848,6 +853,17 @@ export class SessionContext implements ISessionContext {
       this._initStarted.resolve(void 0);
     }
 
+    // If we already have a session, just change the kernel.
+    if (this._session && !this._isTerminating) {
```

Comment:
Is there any side-effect to expect, now that the order of this condition has changed here?

Reply:
The desired effect was that we don't DELETE and POST, but instead PATCH to change kernels.  I confirmed this in my testing.